streamBMSData.c

```diff
@@ -0,0 +1,123 @@
+#include "streamBMSData.h"
+#include "stdbool.h"
+#include "stdlib.h"
+
+void printOnConsole(int temp[], int soc[])
+{
+    int data = 0;
+    printf("\nTemperature Array\n");
+    for(data =0;data < 50; data ++)
+    {
+        printf("%d ,", temp[data]);
+    }
+    printf("\nSOC Array\n");
+    for(data =0; data < 50; data ++)
+    {
+        printf("%d ,", soc[data]);
+    }
+    
+    printf("\n");
+}
+bool checkThresholdLimit(int minVal, int maxVal, int temp)
+{
+    if(temp >= minVal  && temp <= maxVal)
+    {
+        return true;
+    }
+    return false;
+}
+
+bool checkTempinRange(int temp)
+{
+    bool isTempInRange;
+    isTempInRange = checkThresholdLimit(-20, 50, temp);
+    return isTempInRange;
+}
+
+
+bool checkSOCinRange(int soc)
+{
+    if(soc >= 40  && soc <= 90)
+    {
+        return true;
+    }
+    return false;
+}
```

Comment:
Liked the way the Single Responsibility Principle is made used👍

Reply:
Thanks